### PR TITLE
Add option for ign override

### PR DESCRIPTION
**Pull request by baude**
Users suggested we add the ability to merge a user's ignition file with
the ignition fie created by podman init.  You can now specify one or
more override files which will be added to the default ignition file.

Fixes: #11383

[NO NEW TESTS NEEDED]

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
A couple of concerns on this PR.  The first is that when ignition merges files, it is considered more of an add than an override, meaning the fields and values in the default ignition are not replaced.  I think this is probably acceptable but I have used the term override for the command line option.  I do not have a better term, but it really is not an override.

File: pkg/machine/qemu/machine.go [hunk 1]
```diff
@@ -8,13 +8,16 @@ import (
...
+	"net/url"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"strconv"
 	"strings"
 	"time"
 
+	"github.com/vincent-petithory/dataurl"
+
```
Comment (rhatdan) [inline pkg/machine/qemu/machine.go:20]: nit remove blank line.

File: docs/source/markdown/podman-machine-init.1.md [hunk 2]
```diff
@@ -32,6 +32,12 @@ Number of CPUs.
 
 Size of the disk for the guest VM in GB.
 
+#### **--ignition-override-path**
+
+A path to a complete or partial ignition file that will be merged with the ignition
+file created with `podman init`.  The merging of fields and values are to add them rather
+than replace the original entries. More than one override may be specified.
+
 #### **--ignition-path**
```
Comment (rhatdan) [inline docs/source/markdown/podman-machine-init.1.md:41]: Why not make ignition-path a slice and allow users to specify it multiple times?

Comment (baude) [reply docs/source/markdown/podman-machine-init.1.md:41]: @rhatdan I like this idea ... thanks for suggesting it.

File: pkg/machine/config.go [hunk 3]
```diff
@@ -14,16 +14,17 @@ import (
...
-	IsDefault    bool
-	Memory       uint64
-	Name         string
-	TimeZone     string
-	URI          url.URL
-	Username     string
+	CPUS                  uint64
+	DiskSize              uint64
+	IgnitionPath          string
+	IgnitionOverridePaths []string
```
Comment (TomSweeneyRedHat) [inline pkg/machine/config.go:20]: To keep my OCD happy, please flip these lines, 'IgnitionO*, should could before "IgnitionP*"

**Comment by baude** [comment]
After more consideration of this and the reported issue, I am going to close this piece of work.  The problem with merging overrides is programmatically we do not know what is a replace vs addition.  For example, if you wanted to add a new directory and we did a replace merge, all other directory based stuff in the default ign file would be replaced.  It would work if we did an add merge.  However, something like ssh keys, most people would prefer an override over an add merge.

The certs issue is being handled by another PR.

**Pull request closed without merging**